src/java.base/share/classes/jdk/internal/foreign/abi/fallback/FallbackLinker.java

```diff
@@ -283,4 +293,41 @@ private static Object readValue(MemorySegment seg, MemoryLayout layout) {
         assert layout == null;
         return null;
     }
+
+    @Override
+    public Map<String, MemoryLayout> canonicalLayouts() {
+        return CANONICAL_LAYOUTS;
```

Comment:
`CANONICAL_LAYOUTS` is set to a `HashMap`, which is not unmodifiable.
```suggestion
        return Collections.unmodifiableMap(CANONICAL_LAYOUTS);
```